modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/service/impl/CommerceDiscountLocalServiceImpl.java

```diff
@@ -1977,6 +1981,26 @@ private void _validate(
 
 			throw new CommerceDiscountMaxPriceValueException();
 		}
+
+		if (commerceDiscountId > 0) {
+			List<CommerceDiscountRule> commerceDiscountRules =
+				_commerceDiscountRuleLocalService.getCommerceDiscountRules(
+					commerceDiscountId, QueryUtil.ALL_POS, QueryUtil.ALL_POS,
+					null);
+
+			for (CommerceDiscountRule commerceDiscountRule :
+					commerceDiscountRules) {
+
+				String type = commerceDiscountRule.getType();
+
+				if (type.equals(
```

Comment:
If you do
`CommerceDiscountRuleConstants.TYPE_CART_TOTAL.equals(commerceDiscountRule.getType())`
you don't need the `type` variable